### PR TITLE
Add PowerShell and YAML linters

### LINKED ISSUE #742: Update YAML files per linter
# 💡 Summary #

What is the work, as a high-level summary?

Update the YAML files to comply with the new YAML linter.

## Motivation and context ##

Why does this work belong in this project?

Linting keeps files consistent.

## Implementation notes ##

Please provide details for implementation, such as:

- an example for how this would be used

Added a new workflow for MegaLinter for YAML.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] The linter no longer complains about YAML files.

### LINKED ISSUE #743: Update PowerShell files per linter
# 💡 Summary #

What is the work, as a high-level summary?

Update PowerShell files to comply with the new PowerShell linter.

## Motivation and context ##

Why does this work belong in this project?

Linting keeps files consistent.

## Implementation notes ##

Please provide details for implementation, such as:

Added a new workflow for MegaLinter for PS.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] The linter no longer complains about PS files.

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

Added 2 new workflows, both of which use MegaLinter to lint.  One lints PowerShell and the other YAML.

## 💭 Motivation and context ##

Linting keeps files consistent.

Closes #742 
Closes #743 

## 🧪 Testing ##

Ran the pipelines and checked the results.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: .github/workflows/run_linter_powershell.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    paths:
+      - '**.ps1'
+      - '**.psm1'
+      - '**.psd1'
+      - '**.pssc'
+      - '**.psrc'
+      - '**.ps1xml'
+      - '**.cdxml'
+    branches:
+      - 700-add-powershell-linter
```
Comment (crutchfield) [inline .github/workflows/run_linter_powershell.yaml:18]: Remove branch - would not trigger other branches (i.e., main, flipper, ...)
You could add this workflow to push path to get a trigger if updating this file on a push

File: .github/workflows/run_linter_yaml.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Purpose: Run the linter against YAML files
+
+name: Lint Check for YAML
+
+# Run whenever YAML files are checked in
+on:
+  push:
+    paths:
+      - '**.yml'
+      - '**.yaml'
```
Comment (crutchfield) [inline .github/workflows/run_linter_yaml.yaml:11]: consider adding workflow yaml to path as well

File: .github/workflows/run_linter_yaml.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Setup Config File
+        run: cp Testing/Linting/MegaLinter/.mega-linter-yaml.yml .mega-linter.yml
+        if: always()
```
Comment (crutchfield) [inline .github/workflows/run_linter_yaml.yaml:26]: Not sure this is being used correctly.  This causes the step to always execute  and return true (even if cancelled). 
Consider something like:
`if: ${{ !cancelled() }}`

File: .github/workflows/run_linter_yaml.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Setup Lint File
+        run: cp Testing/Linting/MegaLinter/.yamllint.yml .yamllint.yml
+        if: always()
+      - name: Lint YAML
+        uses: oxsecurity/megalinter/flavors/ci_light@v7.6.0
+        if: always()
```
Comment (crutchfield) [inline .github/workflows/run_linter_yaml.yaml:32]: As in previous commentm not sure we want always here.  If previous steps are not successful then this is of no use to always run.

File: .github/workflows/run_markdown_check.yaml
Comment (crutchfield) [inline .github/workflows/run_markdown_check.yaml:1]: If you changed this file (not sure why) then suggest you address linter issues (i.e., indentation).

File: .github/workflows/run_opa_tests.yaml
Comment (crutchfield) [inline .github/workflows/run_opa_tests.yaml:1]: Fix indentation

File: .github/workflows/run_psscriptanalyzer.yaml
Comment (crutchfield) [inline .github/workflows/run_psscriptanalyzer.yaml:1]: Maybe we open an issue, PSScriptAnalyzer is for both modules and scripts.  The paths do not cover this properly.

File: .regal/config.yaml
Comment (crutchfield) [inline .regal/config.yaml:1]: Little creep going on here?

File: OPA.ps1
Comment (crutchfield) [inline OPA.ps1:1]: Not sure why this file is here??  Did yo umodify as part of this issue?

**Review by crutchfield — CHANGES_REQUESTED** [review]
Need to revisit PR.  There appear to be a number of files not related to this issues being addressed included in the PR.

File: .github/workflows/run_linter_yaml.yaml (hunk 3 above)
Comment (james-garriss) [reply .github/workflows/run_linter_yaml.yaml:26]: Those are running as intended, or rather they were.  They were are holdover from when both linters were run in the same workflow.  As they are no longer necessary, removed.

File: .github/workflows/run_linter_yaml.yaml (hunk 2 above)
Comment (crutchfield) [reply .github/workflows/run_linter_yaml.yaml:11]: Nope.

File: .github/workflows/run_markdown_check.yaml
Comment (james-garriss) [reply .github/workflows/run_markdown_check.yaml:1]: I did not edit these, but they came through a merge from flipper.

File: .github/workflows/run_opa_tests.yaml
Comment (james-garriss) [reply .github/workflows/run_opa_tests.yaml:1]: I did not edit this; it came through a merge from flipper.

File: .regal/config.yaml
Comment (james-garriss) [reply .regal/config.yaml:1]: I did not edit this; it came through a merge from flipper.

File: OPA.ps1
Comment (james-garriss) [reply OPA.ps1:1]: I did not edit this; it came through a merge from flipper.

File: .github/workflows/run_psscriptanalyzer.yaml
Comment (james-garriss) [reply .github/workflows/run_psscriptanalyzer.yaml:1]: Done.  https://github.com/cisagov/ScubaGear/issues/759

File: .github/workflows/run_linter_powershell.yaml (hunk 1 above)
Comment (james-garriss) [reply .github/workflows/run_linter_powershell.yaml]: This was for testing purposes.  Removed.

File: .github/workflows/run_linter_powershell.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Setup Config File
+        run: cp Testing/Linting/MegaLinter/.mega-linter-powershell.yml .mega-linter.yml
+      - name: Lint PowerShell
+        uses: oxsecurity/megalinter/flavors/dotnet@v7.6.0
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/run_linter_powershell.yaml]: Updated to use latest.  Done.

**Comment by james-garriss** [comment]
Change to ML container version (moved to latest) completed.

**Comment by dagarwal-mitre** [comment]
Considering changing the Linter to Prettier since many of us use the prettier extension already.

**Comment by james-garriss** [comment]
I will create an issue to explore using Prettier.

**Review by dagarwal-mitre — APPROVED** [review]

**Comment by james-garriss** [comment]
@nanda-katikaneni , this PR is ready to review and merge.

**Pull request merged**